sse2neon.h

```diff
@@ -869,6 +869,20 @@ FORCE_INLINE __m128 _mm_shuffle_ps_default(__m128 a,
 
 // FORCE_INLINE __m128 _mm_shuffle_ps(__m128 a, __m128 b, __constrange(0,255)
 // int imm)
+#if defined(__clang__)
+#define _mm_shuffle_ps(a, b, imm)                          \
+    ({                                                     \
+         float32x4_t _input1 = vreinterpretq_f32_m128(a);  \
+         float32x4_t _input2 = vreinterpretq_f32_m128(b);  \
+         float32x4_t _shuf =                               \
+              __builtin_shufflevector(_input1, _input2,    \
+                (imm) & 0x3,                               \
+                ((imm) >> 2) & 0x3,                        \
+                (((imm) >> 4) & 0x3) + 4,                  \
+                (((imm) >> 6) & 0x3) + 4);                 \
```

Comment:
What makes some of the index need to plus 4 and the others don't?

Reply:
> These element indices are numbered sequentially starting with the first vector, continuing into the second vector. Thus, if vec1 is a 4-element vector, index 5 would refer to the second element of vec2. 

The +4 selects from the second vector.

Reply:
Got it. I misread the function of `_mm_shuffle_ps`.